jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/HiveUrlUtils.java

```diff
@@ -0,0 +1,156 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.dataproc.jdbc;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Helper methods to establish Connection for DataprocDriver. */
+public class HiveUrlUtils {
+    public static final String HIVE_DEFAULT_DATABASE = "";
+    public static final int HIVE_DEFAULT_PORT = 10001;
+    public static final String URL_JDBC_PREFIX = "jdbc:";
+
+    /**
+     * Parses client url and extracts Hive connection parameters.
+     *
+     * @param url verified url passed in by client, already of valid format possible formats:
+     *     jdbc:dataproc://hive/<dbName>;clusterName=<>;projectId=<>;region=<>;port=<>
+     *     jdbc:dataproc://hive/<dbName>;clusterName=<>;projectId=<>;region=<>;clusterPoolLabel=foo=bar:env=staging;
+     * @return HiveJdbcConnectionParam object with fields correctly set
+     * @throws IOException
+     * @throws InvalidURLException
+     */
+    public static HiveJdbcConnectionOptions parseHiveUrl(String url) throws InvalidURLException {
+        HiveJdbcConnectionOptions.Builder paramBuilder = HiveJdbcConnectionOptions.builder();
+
+        // Stripped out for the purpose of using URI
+        String urlStripped = url.substring(URL_JDBC_PREFIX.length());
+        URI jdbcURI = URI.create(urlStripped);
+
+        // getPath() will return: /<dbName>;<sessionConfs>, substring gets rid of the "/"
+        String dbAndSessionConfs = jdbcURI.getPath().substring(1);
+        // getQuery() will return the info after '?'
+        String hiveConfs = jdbcURI.getQuery();
+        // getFragmeng() will return the info after '#'
+        String hiveVars = jdbcURI.getFragment();
+
+        String[] params = dbAndSessionConfs.split(";");
+        // Incorrect database name provided
+        if (params.length != 0 && params[0].contains("=")) {
+            throw new InvalidURLException(
+                    String.format(
+                            "Error: %s \n" + "Database name must not contain '='. " +
+                                    "If no dbName is specified, format should be: %s",
+                            params[0], "jdbc:dataproc://hive/;clusterName=<clusterName>"));
+        }
+
+        paramBuilder.setDbName(params.length == 0 ? HIVE_DEFAULT_DATABASE : params[0]);
+        // Params that do not need parsing
+        paramBuilder.setHiveConfs(hiveConfs);
+        paramBuilder.setHiveVars(hiveVars);
+
+        // Parse sessionConfigs to get cluster info
+        List<String> otherSessionConfsBuilder = new ArrayList<>();
+        String projectId = null;
+        String region = null;
+        String clusterPoolLabel = null;
+        String clusterName = null;
+        int port = HIVE_DEFAULT_PORT;
+        for (int i = 1; i < params.length; i++) {
```

Comment:
It could be better to convert this array to map first and after that get properties from it instead of using `switch` statements to achieve this.

Reply:
I've discussed Karthik about this, and decided to keep the switch statement since `clusterName, projectId, clusterPoolLabel, region` are not properties needed by `HiveJdbcConnectionOptions`, and need to be set as properties for another class (in next pr) to get `host` information. Sorry for writing these haven't used parameters early, will pay attention in next pr.

Reply:
I think that it still will work with a map the only difference will be that instead of assigning variables names in `switch` statement you will retrieve them from the map:

```java
Map<String, String> paramsMap = paramsToMap(params);
String projectId = params.get("projectId");
String region = params.get("region");
String clusterPoolLabel =  params.get("clusterPoolLabel");
String clusterName =  params.get("clusterName");
```

After this you can use these values the same way as after `switch` statement now.